File: api_bus/core/models/persona.py
from django.db import models

class Persona(models.Model):
    GENERO_CHOICES = [
        ('M', 'Masculino'),
        ('F', 'Femenino'),
        ('O', 'Otro'),
    ]
    
    TIPO_CHOICES = [
        ('usuario', 'Usuario'),
        ('acompanante', 'Acompañante'),
        ('externo', 'Externo'),
    ]
    
    nombre = models.CharField(max_length=100)
    primer_apellido = models.CharField(max_length=100)
    segundo_apellido = models.CharField(max_length=100, blank=True, null=True)
    genero = models.CharField(max_length=1, choices=GENERO_CHOICES)
    documento_identidad = models.CharField(max_length=20, unique=True)
    fecha_nacimiento = models.DateField()
    tipo = models.CharField(max_length=11, choices=TIPO_CHOICES)
    
    def __str__(self):
        return f"{self.nombre} {self.primer_apellido}"

File: api-v2/models/persona.js
// models/persona.js
import { DataTypes } from 'sequelize';
import sequelize from '../config/db.js';

const GENERO = ['M', 'F', 'O'];
const TIPO = ['usuario', 'acompanante', 'externo', 'trabajador', 'administrador'];

const Persona = sequelize.define(
  'Persona',
  {
    id: { type: DataTypes.BIGINT.UNSIGNED, autoIncrement: true, primaryKey: true },
    nombre: { type: DataTypes.TEXT, allowNull: true },
    primer_apellido: { type: DataTypes.TEXT, allowNull: true },
    segundo_apellido: { type: DataTypes.TEXT, allowNull: true },
    genero: { type: DataTypes.STRING(1), allowNull: true, validate: { isIn: [GENERO] } },
    fecha_nacimiento: { type: DataTypes.DATEONLY, allowNull: true },
    tipo: { type: DataTypes.TEXT, allowNull: true, validate: { isIn: [TIPO] } },
    documento_identidad: { type: DataTypes.TEXT, allowNull: true },
  },
  { tableName: 'Personas', timestamps: true, underscored: true }
);

// 🔑 ASOCIACIONES (inversa hacia Usuario)
Persona.associate = (models) => {
  Persona.hasOne(models.Usuario, { as: 'usuario', foreignKey: 'personaId' });
};

export default Persona;
